code/modules/power/solar.dm

```diff
@@ -56,6 +62,14 @@
 	appearance_flags = TILE_BOUND
 	blocks_emissive = EMISSIVE_BLOCK_UNIQUE
 
+/obj/machinery/power/solar/proc/update_day_night_exposure(datum/source, light_color, light_alpha)
+	SIGNAL_HANDLER
+
+	if(light_alpha < SOLAR_ALPHA_MINIMUM_TO_GENERATE_POWER)
+		light_level_too_low = TRUE
+	else
+		light_level_too_low = FALSE
```

Comment:
You could make this `light_level_too_low = light_alpha < SOLAR_ALPHA_MINIMUM_TO_GENERATE_POWER` if you wanted

Reply:
done!